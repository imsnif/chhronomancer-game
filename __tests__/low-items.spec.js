import 'react-native'
import React from 'react'
import LowItems from '../components/low-items'
import renderer from 'react-test-renderer'

test('LowItems (with no items) => renders correctly', () => {
  const items = [
    { name: 'assistPrevent', fill: false, bright: false },
    { name: 'steal', fill: false, bright: false },
    { name: 'reset', fill: false, bright: false }
  ]
  const tree = renderer.create(
    <LowItems items={items} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('LowItems (with filled items) => renders correctly', () => {
  const items = [
    { name: 'assistPrevent', fill: true, bright: false },
    { name: 'steal', fill: true, bright: false },
    { name: 'reset', fill: true, bright: false }
  ]
  const tree = renderer.create(
    <LowItems items={items} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('LowItems (with bright items) => renders correctly', () => {
  const items = [
    { name: 'assistPrevent', fill: true, bright: true },
    { name: 'steal', fill: true, bright: true },
    { name: 'reset', fill: true, bright: true }
  ]
  const tree = renderer.create(
    <LowItems items={items} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('LowItems (with fill bright mix) => renders correctly', () => {
  const items = [
    { name: 'assistPrevent', fill: true, bright: true },
    { name: 'steal', fill: false, bright: false },
    { name: 'reset', fill: true, bright: false }
  ]
  const tree = renderer.create(
    <LowItems items={items} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
