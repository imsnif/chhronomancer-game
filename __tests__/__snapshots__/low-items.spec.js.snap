// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LowItems (with bright items) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
</View>
`;

exports[`LowItems (with fill bright mix) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
</View>
`;

exports[`LowItems (with filled items) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
</View>
`;

exports[`LowItems (with no items) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "height": 16,
        "margin": 4,
        "width": 16,
      }
    }
  />
</View>
`;
